### PR TITLE
fix: [SRM-11594]: Fetch records for dynatrace is not functional

### PR DESCRIPTION
**Pull request by deepesh-ui**
### Summary
Fixes for : 
1. Fetch records for dynatrace is not functional
2. In verify step On generating new monitored service and adding HealthSource. HealthSourceList was populating as undefined

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 3 Issues 
* Code issue 3 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10780)

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/MonitoredService/MonitoredService.tsx [hunk 1]
```diff
@@ -127,15 +128,14 @@ export default function MonitoredService({
 
   const createMonitoredService = useCallback(async () => {
     const createdMonitoredService = await createDefaultMonitoredService()
+    const { name = '', identifier = '', sources } = createdMonitoredService?.resource?.monitoredService || {}
```
Comment (harshilbits) [inline src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/MonitoredService/MonitoredService.tsx:131]: sources can be defaulted as empty array.

File: src/modules/85-cv/pages/health-source/connectors/Dynatrace/components/DynatraceCustomMetrics/DynatraceCustomMetrics.tsx [hunk 2]
```diff
@@ -207,7 +207,11 @@ export default function DynatraceCustomMetrics(props: DynatraceCustomMetricsProp
                   error={timeseriesDataErrorMessage}
                   sampleData={timeseriesData}
                   queryValue={metricValues.metricSelector}
-                  isQueryExecuted={shouldShowChart}
+                  isQueryExecuted={
+                    shouldShowChart &&
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/connectors/Dynatrace/components/DynatraceCustomMetrics/DynatraceCustomMetrics.tsx:211]: Please add these conditions inside a function and we also unit test these.

File: src/modules/85-cv/pages/health-source/connectors/Dynatrace/DynatraceHealthSource.utils.tsx [hunk 3]
```diff
@@ -69,7 +72,7 @@ export const mapDynatraceMetricDataToHealthSource = (dynatraceMetricData: Dynatr
   }
 }
 export const mapHealthSourceToDynatraceMetricData = (sourceData: any): DynatraceMetricData => {
-  const healthSource: UpdatedHealthSource = sourceData.healthSourceList.find(
+  const healthSource: UpdatedHealthSource = sourceData.healthSourceList?.find(
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/connectors/Dynatrace/DynatraceHealthSource.utils.tsx:75]: null check can also be applied at `sourceData?.healthSourceList`

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/MonitoredService/MonitoredService.tsx (hunk 1 above)
Comment (deepesh-ui) [reply src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/MonitoredService/MonitoredService.tsx:131]: sources is an Object having HealthSource and ChangeSource

File: src/modules/85-cv/pages/health-source/connectors/Dynatrace/DynatraceHealthSource.utils.tsx (hunk 3 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/Dynatrace/DynatraceHealthSource.utils.tsx]: done

File: src/modules/85-cv/pages/health-source/connectors/Dynatrace/components/DynatraceCustomMetrics/DynatraceCustomMetrics.tsx (hunk 2 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/Dynatrace/components/DynatraceCustomMetrics/DynatraceCustomMetrics.tsx]: done

**Review by harshilbits — APPROVED** [review]

**Review by amitsh14 — APPROVED** [review]

**Pull request merged**